integrations/jira/jira/client.py

```diff
@@ -0,0 +1,116 @@
+import base64
+import typing
+
+import httpx
+from loguru import logger
+
+from jira.overrides import JiraResourceConfig
+from port_ocean.context.event import event
+
+PAGE_SIZE = 50
+WEBHOOK_NAME = "Port-Ocean-RT-Webhook"
+
+WEBHOOK_EVENTS = [
+    "jira:issue_created",
+    "jira:issue_updated",
+    "jira:issue_deleted",
+    "project_created",
+    "project_updated",
+    "project_deleted",
+    "project_soft_deleted",
+    "project_restored_deleted",
+    "project_archived",
+    "project_restored_archived",
+]
+
+
+class JiraClient:
+    def __init__(self, jira_url, jira_email, jira_token) -> None:
+        self.jira_url = jira_url
+        self.jira_rest_url = f"{self.jira_url}/rest"
+        self.jira_email = jira_email
+        self.jira_token = jira_token
+
+        auth_message = f"{self.jira_email}:{self.jira_token}"
+        auth_bytes = auth_message.encode("ascii")
+        b64_bytes = base64.b64encode(auth_bytes)
+        b64_message = b64_bytes.decode("ascii")
+        auth_value = f"Basic {b64_message}"
+
+        self.base_headers = {"Authorization": auth_value}
+
+        self.api_url = f"{self.jira_rest_url}/api/3"
+        self.webhooks_url = f"{self.jira_rest_url}/webhooks/1.0/webhook"
+
+        self.client = httpx.AsyncClient(headers=self.base_headers)
+
+    async def create_real_time_updates_webhook(self, app_host: str):
+        # webhook_target_app_host = f"{app_host}/integration/webhook"
+        # webhook_target_app_host = f"{app_host}"
+        webhook_target_app_host = "https://smee.getport.io/WIxEj7z8VIyVbUid"
+        webhook_check_response = await self.client.get(f"{self.webhooks_url}")
+        webhook_check_response.raise_for_status()
+        webhook_check = webhook_check_response.json()
+
+        for webhook in webhook_check:
+            if webhook["url"] == webhook_target_app_host:
+                logger.info("Ocean real time reporting webhook already exists")
+                return
+
+        body = {
+            "name": WEBHOOK_NAME,
```

Comment:
Think about connecting the integration id

User might have multiple jira integrations in the future

Reply:
Added integration identifier at the start, concat with a hyphen (`-`)